module/modmanager/manager.go

```diff
@@ -442,6 +447,11 @@ func (mgr *Manager) Reconfigure(ctx context.Context, conf config.Module) ([]reso
 		return nil, errors.Errorf("cannot reconfigure module %s as it does not exist", conf.Name)
 	}
 
+	exists, existingName := mgr.execPathAlreadyExists(&conf)
+	if exists {
+		return nil, errors.Errorf("An existing module %s already exists with the same executable path as module %s", existingName, conf.Name)
```

Comment:
this is happening in reconfigure. thinking through the broader flow here:
- I create module 1
- I create module 2 with same exe path
- module 2 fails to `add()`
- now I reconfigure module 1
- does it error, leaving zero working copies? or is module 2 not in `mgr.modules`, because it failed to `add()`?

Reply:
Good question. the 2nd module will never be added to mgr.modules in add(). An error is returned during this validation before it is attempted to be started and added to mgr.modules.

So in the case you brought up, module 1 will be added, module 2 will not, when module 1 is reconfigured it will pass this validation because it has no awareness that module 2 is failing to be added.

Reply:
I also did a another round of manual testing and ensured this is the case. Didn't see this issue or anything else in the various scenarios I tested.